plan/task/exec.go

```diff
@@ -150,19 +153,46 @@ func (t *stopAsyncExecTask) Run(ctx context.Context, pctx *plancontext.Context,
 		return nil, err
 	}
 
+	timeout, err := v.Lookup("timeout").Int64()
+	if err != nil {
+		return nil, err
+	}
+
 	lg := log.Ctx(ctx)
 
-	exitCode, err := s.StopContainer(ctx, ctrID)
+	exitCode, err := s.StopContainer(ctx, ctrID, time.Duration(timeout))
 	if err != nil {
-		return nil, fmt.Errorf("failed to stop async exec %s: %w", ctrID, err)
+		return nil, fmt.Errorf("failed to stop exec %s: %w", ctrID, err)
 	}
-	lg.Debug().Msgf("stopped async exec %s", ctrID)
+	lg.Debug().Msgf("exec %s stopped with exit code %d", ctrID, exitCode)
 
 	return compiler.NewValue().FillFields(map[string]interface{}{
 		"exit": exitCode,
 	})
 }
 
+type sendSignalTask struct {
+}
+
+func (t *sendSignalTask) Run(ctx context.Context, pctx *plancontext.Context, s *solver.Solver, v *compiler.Value) (*compiler.Value, error) {
+	ctrID, err := v.LookupPath(cue.MakePath(cue.Str("input"), cue.Hid("_id", pkg.DaggerPackage))).String()
+	if err != nil {
+		return nil, err
+	}
+
+	sigVal, err := v.Lookup("signal").Int64()
+	if err != nil {
+		return nil, err
+	}
+	sig := syscall.Signal(sigVal)
+
+	if err := s.SignalContainer(ctx, ctrID, sig); err != nil {
+		return nil, fmt.Errorf("failed to send signal %d to exec %s: %w", sig, ctrID, err)
+	}
+
+	return compiler.NewValue(), nil
```

Comment:
Why not just return nil since the value is unused after that? or v?

Reply:
I think I tried returning just `nil` but got a nil pointer exception when the caller of this method tried to unify it with the original value. I guess you could have `v` unify with itself too but not sure there's much difference between that and returning a new empty value here.

Reply:
Okay, that's strange but it's ok to just return a new value then, thank you!

Reply:
Not strange, it's what I'd do. You can see it in `Nop` and `ClientFilesystemWrite` for example.

You need to return a value and it's wasteful to re-fill the "input" value. Tasks only return new values.